packages/react/src/ReactRenderer.tsx

```diff
@@ -120,11 +148,27 @@ export class ReactRenderer<R = unknown, P extends Record<string, any> = object>
     const props = this.props
     const editor = this.editor as EditorWithContentComponent
 
-    if (isClassComponent(Component) || isForwardRefComponent(Component)) {
-      // @ts-ignore This is a hack to make the ref work
-      props.ref = (ref: R) => {
-        this.ref = ref
+    // Handle ref forwarding with React 18/19 compatibility
+    const isReact19 = isReact19Plus()
+    const isClassComp = isClassComponent(Component)
+    const isForwardRefComp = isForwardRefComponent(Component)
+
+    if (!props.ref) {
+      if (isReact19) {
+        // React 19: ref is a standard prop for all components
+        // @ts-ignore - Setting ref prop for React 19 compatibility
+        props.ref = (ref: R) => {
+          this.ref = ref
+        }
+      } else if (isClassComp || isForwardRefComp) {
+        // React 18 and prior: only set ref for class components and forwardRef components
+        // @ts-ignore - Setting ref prop for React 18 class/forwardRef components
+        props.ref = (ref: R) => {
+          this.ref = ref
+        }
       }
+      // For function components in React 18, we can't use ref - the component won't receive it
+      // This is a limitation we have to accept for React 18 function components without forwardRef
```

Comment:
Should we throw an error or log a warning in this case? If the developer tries to send a ref but it's a React 18 function component.

Reply:
Not sure as this is only used as a function to check if we should move forward with the ref creation or not so throwing or logging could be unwanted here.